src/dmd/tokens.d

```diff
@@ -1094,3 +1072,64 @@ nothrow:
     }
 }
 
+/**
+ * Write a character, using a readable escape sequence if needed
+ *
+ * Useful for printing "" string literals in e.g. error messages, ddoc, or the `.stringof` property
+ *
+ * Params:
+ *   buf = buffer to append character in
+ *   c = code point to write
+ */
+nothrow
+void writeCharLiteral(ref OutBuffer buf, dchar c)
```

Comment:
It might be worth making sure this gets inlined

Reply:
Is this performance critical, and is there a reason to distrust the compiler's judgement on whether to inline this function?